### PR TITLE
Subject: 7060x6-64pe 256x100g support

### PR DESCRIPTION
**Pull request by ccroy-arista**
#### Why I did it
Baseline implementation for 256x100g support in the Arista-7060X6-64PE (DCS-7060X6-64PE) has been requested by MSFT.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Added the necessary files for baseline implementation of 256x100g implementation for Quicksilver OSFP.

#### How to verify it
Sanitize DUT with this change and confirm that the relevant interfaces are up.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305
- [x] 202405

#### Tested branch (Please provide the tested image version)
master

#### Description for the changelog
Baseline support for the Arista-7060X6-64PE-256x100G platform variant.

### CODE REVIEW

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/buffer_ports.j2 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{%- macro generate_port_lists(PORT_ALL) %}
```
Comment (r12f) [inline device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/buffer_ports.j2:1]: hi Chris, do you mind to update the HWSKU name according to our naming convention, e.g. C256 for 256x100G?

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/hwsku.json [hunk 2]
```diff
@@ -0,0 +1,202 @@
+{
+    "interfaces": {
+        "Ethernet0": {
+            "default_brkout_mode": "8x100G"
+        },
+        "Ethernet8": {
+            "default_brkout_mode": "2x400G"
```
Comment (r12f) [inline device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/hwsku.json:7]: does this mean the port names are not continuous when in use? e.g. - Ethernet0, Ethernet1, ...,  Ethernet7, Ethernet16, Ethernet17, ... Ethernet23, ....

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/port_config.ini [hunk 3]
```diff
@@ -0,0 +1,323 @@
+# name          lanes                           alias           index   speed   fec
+Ethernet0       17                              Ethernet1/1     1       100000  rs
+Ethernet1       18                              Ethernet1/2     1       100000  rs
+Ethernet2       19                              Ethernet1/3     1       100000  rs
+Ethernet3       20                              Ethernet1/4     1       100000  rs
+Ethernet4       21                              Ethernet1/5     1       100000  rs
+Ethernet5       22                              Ethernet1/6     1       100000  rs
+Ethernet6       23                              Ethernet1/7     1       100000  rs
+Ethernet7       24                              Ethernet1/8     1       100000  rs
+Ethernet8       1,2,3,4                         Ethernet2/1     2       400000  rs
```
Comment (r12f) [inline device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/port_config.ini:10]: is it possible to skip the non-used ports, so we can have continuous port names?

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/hwsku.json (hunk 2 above)
Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/hwsku.json:7]: Correct, here is an example of the output:
```
$ show int sta
  Interface            Lanes    Speed    MTU    FEC         Alias    Vlan    Oper    Admin                           Type    Asym PFC
-----------  ---------------  -------  -----  -----  ------------  ------  ------  -------  -----------------------------  ----------
  Ethernet0               17     100G   9100     rs   Ethernet1/1  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet1               18     100G   9100     rs   Ethernet1/2  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet2               19     100G   9100     rs   Ethernet1/3  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet3               20     100G   9100     rs   Ethernet1/4  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet4               21     100G   9100     rs   Ethernet1/5  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet5               22     100G   9100     rs   Ethernet1/6  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet6               23     100G   9100     rs   Ethernet1/7  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet7               24     100G   9100     rs   Ethernet1/8  routed      up       up  OSFP 8X Pluggable Transceiver         off
  Ethernet8          1,2,3,4     400G   9100    N/A   Ethernet2/1  routed    down     down                            N/A         off
 Ethernet12          5,6,7,8     400G   9100    N/A   Ethernet2/5  routed    down     down                            N/A         off
 Ethernet16                9     100G   9100     rs   Ethernet3/1  routed      up       up  OSFP 8X Pluggable Transceiver         off
 Ethernet17               10     100G   9100     rs   Ethernet3/2  routed      up       up  OSFP 8X Pluggable Transceiver         off
...
```

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/buffer_ports.j2 (hunk 1 above)
Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/buffer_ports.j2:1]: Hi Riff, I will update the name.

Can you confirm this result will be sufficient: Arista-7060X6-64PE-C256

Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/buffer_ports.j2]: @r12f we also need to factor in the two 10G SFP ports. That would make the name: `Arista-7060X6-64PE-C256S2`
Please confirm that is acceptable.

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/port_config.ini (hunk 3 above)
Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/port_config.ini]: @r12f I will explore options and get back to you.

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/buffer_ports.j2 (hunk 1 above)
Comment (r12f) [reply device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/buffer_ports.j2]: yea, as discussed offline, the sku name looks great to me!

**Comment by r12f** [comment]
Chris to update this, early this week.

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/port_config.ini (hunk 3 above)
Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/port_config.ini]: As discussed in yesterday's meeting, we will forego contiguous port names in light of the platform.json complications it raises, marking this as resolved.

File: device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/hwsku.json (hunk 2 above)
Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x100G/hwsku.json]: Down ports removed in latest push to PR; marking resolve.

**Review by r12f — APPROVED** [review]
lgtm! thanks Chris!

**Pull request merged**